### ISSUE TITLE
Dune-build-info reports wrong library version

### ISSUE THREAD

**Issue by reynir** [ocaml/dune#8025]
This was observed in https://github.com/mirage/mirage/issues/1438

## Expected Behavior
ocplib-endian's version is reported as 1.2, or unknown.

## Actual Behavior
ocplib-endian's version is reported to be the short git hash of the git repository in CWD, unrelated to ocplib-endian.

## Additional information 
It is unclear to me what a version of a library is as understood by dune-build-info. It also doesn't report the version of the compiler.

**Comment by emillon** [comment]
> It is unclear to me what a version of a library is as understood by dune-build-info

An explanation of the behavior can be found [here](https://dune.readthedocs.io/en/stable/dune-libs.html#dune-build-info-library): it looks for `(version)` in `dune-project` (this field is added by `dune-release` at release time or `dune subst` at pin time), or calls `git describe` otherwise.

Since the `dune-project` file in the ocplib-endian.1.2 tarball does not have a `(version)` field, you're hitting the second case. If you vendored the sources, it will call the VCS of your project.

**Comment by reynir** [comment]
Yes, thanks, I read that section. What was unclear to me was what it does when the library doesn't come from a development library and doesn't have `(version)` in `dune-project`. I realize now that it's `None` then.

What confused me is I didn't consider `ocplib-endian` vendored, but I guess in a sense it is due to opam-monorepo (through the mirage tool) fetches the source of ocplib-endian and puts it in `duniverse/`. The version fetched has very little to do with the git repository and its commit hash, however.

Would it make sense to check if the `dune-project` file inspected is checked in with the version control system? If it's not then the version obtained from the VCS is easily misleading.

**Comment by emillon** [comment]
I think we might also do something in the case where the project is in a vendored dir. It doesn't seem useful to propagate version info in that case.

**Comment by emillon** [comment]
We discussed that at this week's dev meeting and we agree to return `None` in that case.

**Comment by reynir** [comment]
Thanks, that sounds great :)
